### ISSUE TITLE
SendText+ to SublimeREPL Python doesn't work

### ISSUE THREAD

**Issue by pingshunhuangalex** [randy3k/SendTextPlus#24]
Hi Randy

I just figured out that SublimeREPL hasn't been updating for more than a year, and the Eval/Transfer function is completely down on ST3 (I'm using python 3.5.1 on win10 64 bit).

Having read the issue question in SublimeREPL one by one, a thread led me to SendText+ which seems the perfect solution for my situation. I installed it and set SublimeREPL as the default program. Now when I execute a simple test script using ctrl+enter. I got this error message:

> Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 1
    %cpaste -q
    ^
SyntaxError: invalid syntax

The sample code I'm using is:
```
height = 1.75
weight = 80.5

bmi = weight / height ** 2

if bmi < 18.5:
    print('过轻')
elif bmi <= 25:
    print('正常')
elif bmi <= 28:
    print('过重')
elif bmi <= 32:
    print('肥胖')
else:
    print('严重肥胖')
```

Also, when I tried the following command one by one, it works perfectly. However, if I select both of them and ran them at the same time, it just fails...
```
print(1+2)
print(2+3)
```

I also tried the everything above in ipython in SublimeREPL(Jupyter 4.0.3), it gets even worse as nothing showed up no matter how hard I pressed ctrl+enter...

Hope you can help here, I'm pretty desperate now... Thanks.

**Comment by randy3k** [comment]
Just like what I have mentioned in the description, SendText+ works better with R in SublimeREPL. The python part is very buggy and I guess there is no easy way to fix it.

I suppose that you are running a windows OS, on windows, cmder is recommended

**Comment by saraphineshps** [comment]
Can you kindly instruct me on how to use cmder? I presume it's just cmd in windows?

I change the default program to cmder and open cmd, then ctrl-enter the code in sublime text, still nothing happens... Thanks.

**Comment by randy3k** [comment]
I meant [this](http://cmder.net).

**Comment by saraphineshps** [comment]
Sorry it still doesn't work... I install cmder and open it with sublime side by side. Choose the default program as cmder and ctrl+enter the command, nothing gets sent.

**Comment by saraphineshps** [comment]
I kind of understand the error message now. At the end of the loop, SendText+ will add a "--" which then triggers error. Do you know what happened there? The "--" seems to be find when it's not a loop or script block, e.g. a = b

**Comment by randy3k** [comment]
It is added for Python block code in iPython.

**Comment by saraphineshps** [comment]
Hi Randy

When I press ctrl+enter in sublime text. It copies the code just fine, but I have to right click in cmder to paste the code. It doesn't seem to be sent to cmder automatically. Is it intentional or did I do something wrong? Is there a proper order to open sublime and cmder? Thanks.

**Comment by randy3k** [comment]
It should send the code automatically. Did you choose cmder in "SendTextPlus: Choose Program"? Also, you have to change the default keybind for pasting code to "Ctrl+Shift+v" in cmder preference.

**Comment by pingshunhuangalex** [comment]
Thanks for the reply. I've done both of them. Any other possible fix?

**Comment by randy3k** [comment]
Could you manually copy the command and "ctrl+shift+v" (not right click) on cmder console?

**Comment by pingshunhuangalex** [comment]
Yes, it works pretty well this way.

**Comment by randy3k** [comment]
it is odd that it doesn't work if you have set the active program to cmder and changed the paste shortcut...

I assume that there is no error message in sublime console, right?

**Comment by pingshunhuangalex** [comment]
There is no error message. I'm using win10(64-bit). Do you have any computer that is running win10? Is it possible that some win10 protection/authentication is preventing sublime/sendtext+ from sending command to cmder?

**Comment by randy3k** [comment]
It works fine on my windows 10 (64 bit) machine, sublime 3103 and the latest cmder.

**Comment by pingshunhuangalex** [comment]
Me too... Are you opening everything as administrator? If not, I'm really out of thought now.

**Comment by randy3k** [comment]
I open sublime text in regular mode, though my account is an administrator account. SendText+ contains a file [AutoHotKeyU32.exe](https://autohotkey.com) to send fake keyboard keystrokes to cmder, I can only deduce that there is some problems of executing that file on your computer...

**Comment by pingshunhuangalex** [comment]
Ok, I think I replicate everything you listed here... Anyway maybe I should take it from here. Thanks.
Can you tell me where the package is installed? It's not in the installed package...

**Comment by randy3k** [comment]
Not in installed package?? You should see a file called "SendText+.sublime-package" or "SendTextPlus.sublime-package" under the directory "Installed Packages".

**Comment by pingshunhuangalex** [comment]
Sorry I mean "Packages", there is a "SendTextPlus.sublime-package" in the "Installed Packages", but there is no SendTextPlus folder in "Packages" where people can usually see the raw code and key binding stuff.

**Comment by pingshunhuangalex** [comment]
I ran it as administrator, still no luck. It works perfectly for both lines and source file, but just won't be sent automatically to cmder unless I press right button. If you somehow find a possible reason, please don't hesitate to reply me back. Otherwise, thank you for the awesome plugin, I'll still continue to use it :)

**Comment by pingshunhuangalex** [comment]
Hi Randy

Thank goodness, I got it to work. Hope it can help people who are also concerned. Here are the reasons why it was not functioning previously.

1. The file in sublime text has to be open freshly, and it can not be in the form of sublime cache. When I open up a new file, Sendtext+ works... BUT in sublime not in terminal, i.e. it paste the command and code right where I wrote them in sublime. However, that's the time I think I might have a shot on getting this to work.

2. Sublime cannot be set up as a startup task in cmder (open up sublime and ipython side by side both in the cmder window). The moment I start up sublime text and cmder one by one on their own, Sendtext+ kicks in like magic!

Thank you Randy for all the help along the way, and thank you again for the amazing plugin.
